src/otx/core/data/dataset/action.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXActionDataset."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+import cv2
+import torch
+from datumaro import Label
+
+if TYPE_CHECKING:
+    import numpy as np
+    from datumaro.components.dataset_base import DatasetItem
+
+from otx.core.data.dataset.base import OTXDataset
+from otx.core.data.entity.action import ActionClsBatchDataEntity, ActionClsDataEntity
+from otx.core.data.entity.base import ImageInfo
+
+
+class OTXActionClsDataset(OTXDataset[ActionClsDataEntity]):
+    """OTXDataset class for action classification task."""
+
+    def _get_item_impl(self, idx: int) -> ActionClsDataEntity | None:
+        item = self.dm_subset.get(id=self.ids[idx], subset=self.dm_subset.name)
+        # video_data = [frame.data for frame in video]
+        video_data = self._get_video_data(item)
+        img_shape = video_data[0].shape
+
+        label_anns = [ann for ann in item.annotations if isinstance(ann, Label)]
+
+        entity = ActionClsDataEntity(
+            image=video_data,
+            img_info=ImageInfo(
+                img_idx=idx,
+                img_shape=img_shape,
+                ori_shape=img_shape,
+                pad_shape=img_shape,
+                scale_factor=(1.0, 1.0),
+            ),
+            labels=torch.as_tensor([ann.label for ann in label_anns]),
```

Comment:
```suggestion
            image=[],
            video=item.media,
            img_info=ImageInfo(
                img_idx=idx,
                img_shape=img_shape,
                ori_shape=img_shape,
                pad_shape=img_shape,
                scale_factor=(1.0, 1.0),
            ),
            labels=torch.as_tensor([ann.label for ann in label_anns]),
```

1. `Dataset._get_item_impl() => image: []`
2. `Dataset._apply_transform() => image: [tv_tensor.Image, ...] ` obtained from `data_entity.video`

Reply:
Done by 93aed91